src/CloudNative.CloudEvents.Avro/BasicGenericRecordSerializer.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Avro;
+using Avro.Generic;
+using Avro.IO;
+using CloudNative.CloudEvents.Avro.Interfaces;
+using System;
+using System.IO;
+
+namespace CloudNative.CloudEvents.Avro;
+
+/// <summary>
+/// The default implementation of the <see cref="IGenericRecordSerializer"/>.
+/// </summary>
+/// <remarks>
+/// Makes use of the Avro <see cref="DefaultReader"/> and <see cref="DefaultWriter"/>
+/// together with the embedded Avro schema.
+/// </remarks>
+sealed internal class BasicGenericRecordSerializer : IGenericRecordSerializer
```

Comment:
Nit: please reorder to `internal sealed class`

Reply:
Suggestion applied